karapace/protobuf/protobuf_to_dict.py

```diff
@@ -0,0 +1,338 @@
+# -*- coding:utf-8 -*-
+"""
+This module provide a small Python library for creating dicts from protocol buffers
+Module based on code :
+https://github.com/wearefair/protobuf-to-dict
+LICENSE: https://github.com/wearefair/protobuf-to-dict/blob/master/LICENSE
+"""
+
+from dateutil.parser import parse as date_parser
+from frozendict import frozendict
+from google.protobuf.descriptor import FieldDescriptor
+from google.protobuf.message import Message
+from google.protobuf.timestamp_pb2 import Timestamp
+
+import datetime
+import six
+
+__all__ = ["protobuf_to_dict", "TYPE_CALLABLE_MAP", "dict_to_protobuf", "REVERSE_TYPE_CALLABLE_MAP"]
+
+Timestamp_type_name = 'Timestamp'
+
+# pylint: disable=E1101
+
+
+def datetime_to_timestamp(dt):
+
+    ts = Timestamp()
+
+    ts.FromDatetime(dt)
+    return ts
+
+
+def timestamp_to_datetime(ts):
+    dt = ts.ToDatetime()
+    return dt
+
+
+EXTENSION_CONTAINER = '___X'
+
+TYPE_CALLABLE_MAP = frozendict({
+    FieldDescriptor.TYPE_DOUBLE: float,
+    FieldDescriptor.TYPE_FLOAT: float,
+    FieldDescriptor.TYPE_INT32: int,
+    FieldDescriptor.TYPE_INT64: int if six.PY3 else six.integer_types[1],
+    FieldDescriptor.TYPE_UINT32: int,
+    FieldDescriptor.TYPE_UINT64: int if six.PY3 else six.integer_types[1],
+    FieldDescriptor.TYPE_SINT32: int,
+    FieldDescriptor.TYPE_SINT64: int if six.PY3 else six.integer_types[1],
+    FieldDescriptor.TYPE_FIXED32: int,
+    FieldDescriptor.TYPE_FIXED64: int if six.PY3 else six.integer_types[1],
+    FieldDescriptor.TYPE_SFIXED32: int,
+    FieldDescriptor.TYPE_SFIXED64: int if six.PY3 else six.integer_types[1],
```

Comment:
Karapace only supports `python>=3.7`. Python2 reached its [end-of-life for 2 years](https://www.python.org/dev/peps/pep-0373/) and it is no longer receiving bug / security fixes.

```suggestion
    FieldDescriptor.TYPE_INT64: int,
    FieldDescriptor.TYPE_UINT32: int,
    FieldDescriptor.TYPE_UINT64: int,
    FieldDescriptor.TYPE_SINT32: int,
    FieldDescriptor.TYPE_SINT64: int,
    FieldDescriptor.TYPE_FIXED32: int,
    FieldDescriptor.TYPE_FIXED64: int,
    FieldDescriptor.TYPE_SFIXED32: int,
    FieldDescriptor.TYPE_SFIXED64: int,
```